modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java

```diff
@@ -180,23 +184,38 @@ protected void resolveLobs(String typeName, Object[] attributes) throws SQLExcep
 
             return dataTypes;
         }
+        finally
+        {
+            if (resultSet != null)
+            {
+                resultSet.close();
+            }
+        }
     }
 
-    public String getNestedTableType(String typeName) throws SQLException
+    private String getTypeFor(String collectionTypeName) throws SQLException
     {
         String dataType = null;
+        ResultSet resultSet = null;
 
         try (PreparedStatement ps = this.prepareStatement(QUERY_ALL_COLL_TYPES))
         {
-            ps.setString(1, typeName);
+            ps.setString(1, collectionTypeName);
 
-            ResultSet resultSet = ps.executeQuery();
+            resultSet = ps.executeQuery();
 
             while (resultSet.next())
             {
                 dataType = resultSet.getString(ELEM_TYPE_NAME);
             }
         }
+        finally
```

Comment:
use try-with-resources instead of close on finally (as in the preparedStatement).